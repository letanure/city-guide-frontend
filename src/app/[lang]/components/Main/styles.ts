import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: #a2aecd;
  height: 100%;
  width: 100%;
`

export const Title = styled.h1`
  color: #333;
  font-size: 5rem;
  font-weight: 700;
  margin-bottom: 2rem;
`

export const Description = styled.h2`
  font-size: 2rem;
  font-weight: 400;
`
