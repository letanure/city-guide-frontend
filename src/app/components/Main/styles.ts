import styled from 'styled-components'

export const Wrapper = styled.main`
  width: 100%;
  height: 100%;
`

export const Title = styled.h1`
  font-size: 5rem;
  font-weight: 700;
  margin-bottom: 2rem;
`

export const Description = styled.h2`
  font-size: 2rem;
  font-weight: 400;
`
