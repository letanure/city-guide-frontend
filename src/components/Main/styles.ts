import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.main`
  ${() => css`
    background-color: #a2aecd;
    height: 100%;
    width: 100%;

    ${media.lessThan('medium')`
      border: 5px solid red
    `}
  `}
`

export const Wrapper2 = styled.main`
  background-color: #a2aecd;
  height: 100%;
  width: 100%;
`

export const Title = styled.h1`
  color: #333;
  font-size: 5rem;
  font-weight: 700;
  margin-bottom: 2rem;
`

export const Description = styled.h2`
  font-size: 2rem;
  font-weight: 400;
`
