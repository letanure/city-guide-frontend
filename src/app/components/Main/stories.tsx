import type { Meta, StoryObj } from '@storybook/react'
import Main from './'

const meta: Meta<typeof Main> = {
  component: Main,
}

export default meta
type Story = StoryObj<typeof Main>

export const Primary: Story = {
  args: {
    title: 'Boilerplate',
    description: 'TypeScript, ReactJS, NextJS e Styled Components',
  },
}
