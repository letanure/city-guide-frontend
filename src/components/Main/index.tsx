'use client'

import * as S from './styles'

type MainProps = {
  title?: string
  description?: string
}

const Main = ({
  title = 'Boilerplate',
  description = 'TypeScript, ReactJS, NextJS e Styled Components',
}: MainProps) => (
  <S.Wrapper>
    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>
  </S.Wrapper>
)

export default Main
