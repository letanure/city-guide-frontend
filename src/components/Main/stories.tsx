import type { Meta, StoryObj } from '@storybook/react'
import Main from './'

const meta: Meta<typeof Main> = {
  component: Main,
  args: {
    title: 'Boilerplate',
    description: 'TypeScript, ReactJS, NextJS e Styled Components',
  },
}

export default meta

type Story = StoryObj<typeof Main>

export const Default: Story = {}
