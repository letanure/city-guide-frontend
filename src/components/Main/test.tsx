import { render, screen } from '@testing-library/react'
import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /Boilerplate/i }),
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it.skip('should render the colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({
      'background-color': 'rgb(162, 174, 205)',
    })
  })
})
