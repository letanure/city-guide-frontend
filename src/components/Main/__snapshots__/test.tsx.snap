// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  background-color: #a2aecd;
  height: 100%;
  width: 100%;
}

.c1 {
  color: #333;
  font-size: 5rem;
  font-weight: 700;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2rem;
  font-weight: 400;
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    Boilerplate
  </h1>
  <h2
    class="c2"
  >
    TypeScript, ReactJS, NextJS e Styled Components
  </h2>
</main>
`;
